src/hivpy/sex_behaviour_data.py

```diff
@@ -84,12 +81,29 @@ def __init__(self, filename):
                 SexType.Female: self._get_discrete_dist(
                     "initial_sex_behaviour_probabilities", "Female")}
 
+            # risk reduction factors
             self.rred_initial = self.data["rred_initial"]
 
+            self.age_based_risk = np.array(
+                rng.choice(self.data["age_based_risk_options"]["risk_factor"],
+                           p=self.data["age_based_risk_options"]["Probability"]))
+
             self.new_partner_dist = self._get_discrete_dist("new_partner_factor")
 
+            self.rred_long_term_partnered = self._get_discrete_dist("rred_partnered")
+
             self.p_rred_p_dist = self._get_discrete_dist("population_rred_personal")
 
+            self.rred_diagnosis = self._get_discrete_dist("rred_diagnosis")
+            self.rred_diagnosis_period = self.data["rred_diagnosis"]["Period"]
+
+            self.yearly_risk_change = {"1990s": self._get_discrete_dist("yearly_risk_change_90s"),
+                                       "2010s": self._get_discrete_dist("yearly_risk_change_10s")}
```

Comment:
I'm happy to make these numerical if you like, I'm just not sure what you mean by keys that can be processed automatically? I wouldn't have thought there was much difference in the type of key as long as it's hashable?

Reply:
I meant in case we need to find which decade we are in, for instance, and match to the relevant yearly risk. Having the keys as numbers would make it easier to retrieve the relevant entry. But that's probably more trouble than it's worth.